target_postgres/sinks.py

```diff
@@ -314,34 +314,42 @@ def activate_version(self, new_version: int) -> None:
         Args:
             new_version: The version number to activate.
         """
+        if self.config["activate_version"] is False:
+            self.logger.warning(
+                "An activate version message was received, but activate_version is set "
+                "to false so it was ignored."
+            )
+            return
+
         # There's nothing to do if the table doesn't exist yet
         # (which it won't the first time the stream is processed)
         if not self.connector.table_exists(self.full_table_name):
             return
 
         deleted_at = now()
-        # Different from SingerSDK as we need to handle types the
-        # same as SCHEMA messsages
-        datetime_type = self.connector.to_sql_type(
-            {"type": "string", "format": "date-time"}
-        )
-
-        # Different from SingerSDK as we need to handle types the
-        # same as SCHEMA messsages
-        integer_type = self.connector.to_sql_type({"type": "integer"})
 
         with self.connector._connect() as connection, connection.begin():
             if not self.connector.column_exists(
                 full_table_name=self.full_table_name,
                 column_name=self.version_column_name,
                 connection=connection,
             ):
-                self.connector.prepare_column(  # type: ignore[call-arg]
-                    self.full_table_name,
-                    self.version_column_name,  # type: ignore[arg-type]
-                    sql_type=integer_type,
+                raise RuntimeError(
```

Comment:
Yeah I dove in here. Theoretically this error should never be hit because we always create the column, but I like it as a sanity check. If anything changes down the line, the error that would appear otherwise is not as intuitive.